src/leap/mail/mailbox_indexer.py

```diff
@@ -0,0 +1,333 @@
+    try:
+        uuid.UUID(str(mailbox_id))
+    except (AttributeError, ValueError):
+        raise AssertionError(
+            "the mbox_id is not a valid uuid: %s" % mailbox_id)
+
+
+class MailboxIndexer(object):
+    """
+    This class contains the commands needed to create, modify and alter the
+    local-only UID tables for a given mailbox.
+
+    Its purpouse is to keep a local-only index with the messages in each
+    mailbox, mainly to satisfy the demands of the IMAP specification, but
+    useful too for any effective listing of the messages in a mailbox.
+
+    Since the incoming mail can be processed at any time in any replica, it's
+    preferred not to attempt to maintain a global chronological global index.
+
+    These indexes are Message Attributes needed for the IMAP specification (rfc
+    3501), although they can be useful for other non-imap store
+    implementations.
+    """
+    # The uids are expected to be 32-bits values, but the ROWIDs in sqlite
+    # are 64-bit values. I *don't* think it really matters for any
+    # practical use, but it's good to remmeber we've got that difference going
+    # on.
+
+    store = None
+    table_preffix = "leapmail_uid_"
+
+    def __init__(self, store):
+        self.store = store
+
+    def _query(self, *args, **kw):
+        assert self.store is not None
+        return self.store.raw_sqlcipher_query(*args, **kw)
+
+    def create_table(self, mailbox_id):
+        """
+        Create the UID table for a given mailbox.
+        :param mailbox: the mailbox identifier.
+        :type mailbox: str
+        :rtype: Deferred
+        """
+        check_good_uuid(mailbox_id)
+        sql = ("CREATE TABLE if not exists {preffix}{name}( "
+               "uid  INTEGER PRIMARY KEY AUTOINCREMENT, "
+               "hash TEXT UNIQUE NOT NULL)".format(
+                   preffix=self.table_preffix, name=sanitize(mailbox_id)))
+        return self._query(sql)
+
+    def delete_table(self, mailbox_id):
+        """
+        Delete the UID table for a given mailbox.
+        :param mailbox: the mailbox name
+        :type mailbox: str
+        :rtype: Deferred
+        """
+        check_good_uuid(mailbox_id)
+        sql = ("DROP TABLE if exists {preffix}{name}".format(
+            preffix=self.table_preffix, name=sanitize(mailbox_id)))
+        return self._query(sql)
+
+    def rename_table(self, oldmailbox, newmailbox):
+        """
+        Delete the UID table for a given mailbox.
+        :param oldmailbox: the old mailbox name
+        :type oldmailbox: str
+        :param newmailbox: the new mailbox name
+        :type newmailbox: str
+        :rtype: Deferred
+        """
+        assert oldmailbox
+        assert newmailbox
+        assert oldmailbox != newmailbox
+        sql = ("ALTER TABLE {preffix}{old} "
+               "RENAME TO {preffix}{new}".format(
+                   preffix=self.table_preffix,
+                   old=sanitize(oldmailbox), new=sanitize(newmailbox)))
```

Comment:
Hmm ... I'd swear I had deleted this method.
This doesn't make sense now, because mailboxes uuid won't change, even if we rename them.
Removed.